threads/test_thread_rand.c

```diff
@@ -0,0 +1,88 @@
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/threads.h>
+
+#include "unity_fixture.h"
+
+
+typedef struct {
+	uint16_t time;
+	uint16_t id;
+} threads_self_t;
+
+
+static int cthread = 10, fin = 0;
+
+
+static char stack[10 * 512];
+
+
+static void test_thread(void *arg)
+{
+	threads_self_t *self = (threads_self_t *)arg;
+	usleep(1000 * self->time);
+	__sync_fetch_and_add(&fin, 1);
+	endthread();
+}
+
+
+TEST_GROUP(threads_rand);
+
+
+TEST_SETUP(threads_rand)
+{
+}
+
+
+TEST_TEAR_DOWN(threads_rand)
+{
+}
+
+
+TEST(threads_rand, test_1)
+{
+	threads_self_t params[cthread];
+	int ret, joined = 0, i;
+
+	srand(7);
+	/* assign threads parameters */
+	for (i = 0; i < cthread; i++) {
+		params[i].id = i;
+		params[i].time = 100 + rand() % 200; /* assign thread lifespan in milliseconds */
```

Comment:
I think that time randomization range should be defined properly using defined scale projected on RAND_MAX macro.

Reply:
Yes it would be more professional, but i would rather stay with this simple approach with just `%` and added constant value as we only need an integer and we do not really bother with edge cases. This saves us couple of casts (from integer to double and then finally to integer for `sleep` functions) and keeps the code clean.